### PR TITLE
Create Dockerfile for quill + maintenance on other Dockerfiles

### PR DESCRIPTION
**Pull request by mileslane**
### Changes

1. git rm Dockerfile
2. git cp Dockerfile.cvp Dockerfile.quill
3. git mv Dockerfile-cds-tools Dockerfile.cds-tools
4. git mv Dockerfile-chefdk Dockerfile.chefdk
5. ruby 2.7.x for quill
6. Remove the JDK from the quill image

### CODE REVIEW

File: Dockerfile.cvp [hunk 1]
```diff
@@ -24,12 +24,13 @@ RUN apt-get update \
   apt-transport-https \
   build-essential \
   ca-certificates \
-  curl \
+  curl libcurl4-openssl-dev \
```
Comment (olbrich) [inline Dockerfile.cvp:27]: Why is this needed? Also, please keep each package on a single line in alphabetical order.

File: Dockerfile.cvp [hunk 2]
```diff
@@ -24,12 +24,13 @@ RUN apt-get update \
...
   git \
   jq \
   software-properties-common \
   unzip \
   zip \
+  zlib1g-dev \
```
Comment (olbrich) [inline Dockerfile.cvp:33]: What is this for?
Maybe add a comment after the `\` indicating why it was needed.

**Review by olbrich — APPROVED** [review]
Looks good, add the comments and then you will need to let all the images build before github will let you merge it.

File: Dockerfile.cvp (hunk 1 above)
Comment (mileslane) [reply Dockerfile.cvp]: Done and commented.

File: Dockerfile.cvp (hunk 2 above)
Comment (mileslane) [reply Dockerfile.cvp]: Commented.

File: Dockerfile.cds-tools [hunk 3]
```diff
@@ -11,13 +11,19 @@ RUN apt-get install -y \
...
-  libpq-dev \
+  libcurl4-openssl-dev `# Needed to pull in libcurl4 for curl installation and enable to openssl TLS backend` \
+  libmagickcore-dev \
+  libmagickwand-dev \
+  libpq-dev `# Needed to provide libpq-fe.h header to build 'pg' gem native extensions` \
   software-properties-common \
   unzip \
-  zip
+  zip \
+  zlib1g-dev `# Fixes 'zlib is missing; necessary for building libxml2'` \
```
Comment (olbrich) [inline Dockerfile.cds-tools:23]: Does this happen when installing a gem?

File: Dockerfile.chefdk [hunk 4]
```diff
@@ -8,8 +8,10 @@ RUN apt-get update && apt-get install -y \
   curl \
   git \
   jq \
-  libcurl3 libcurl3-gnutls libcurl4-openssl-dev \
-  libpq-dev \
+  libcurl3 \
+  libcurl3-gnutls \
+  libcurl4-openssl-dev `# Needed for curl installation` \
+  libpq-dev `# Needed to provide libpq-fe.h header to build 'pg' gem native extensions` \
```
Comment (olbrich) [inline Dockerfile.chefdk:14]: Does ChefDK even need the `pg` gem?

File: Dockerfile.cds-tools (hunk 3 above)
Comment (mileslane) [reply Dockerfile.cds-tools:23]: I believe it is nokogiri extensions

File: Dockerfile.chefdk (hunk 4 above)
Comment (mileslane) [reply Dockerfile.chefdk:14]: I doubt it, but if the application source is pulled in and there is a bundle install, it'll want to build the gem extensions, right?

File: Dockerfile.cds-tools (hunk 3 above)
Comment (olbrich) [reply Dockerfile.cds-tools:23]: If this is related to nokogiri then maybe this should be
```suggestion
  zlib1g-dev `# For nokogiri -- fixes 'zlib is missing; necessary for building libxml2'` \
```

File: Dockerfile.cds-tools [hunk 5]
```diff
@@ -11,13 +11,19 @@ RUN apt-get install -y \
   curl \
   git \
   gnupg \
-  imagemagick libmagickcore-dev libmagickwand-dev \
+  imagemagick \
```
Comment (olbrich) [inline Dockerfile.cds-tools:14]: I think this relates to the 'rmagick' gem.

File: Dockerfile.cds-tools [hunk 6]
```diff
@@ -11,13 +11,19 @@ RUN apt-get install -y \
...
   jq \
-  libcurl4 libcurl3-gnutls libcurl4-openssl-dev \
-  libpq-dev \
+  libcurl4-openssl-dev `# Needed to pull in libcurl4 for curl installation and enable to openssl TLS backend` \
+  libmagickcore-dev \
+  libmagickwand-dev \
```
Comment (olbrich) [inline Dockerfile.cds-tools:18]: These are also needed for `rmagick`

File: Dockerfile.cds-tools [hunk 7]
```diff
@@ -11,13 +11,19 @@ RUN apt-get install -y \
...
+  libpq-dev `# Needed to provide libpq-fe.h header to build 'pg' gem native extensions` \
```
Comment (olbrich) [inline Dockerfile.cds-tools:19]:
```suggestion
  libpq-dev `# Needed for 'pg' gem` \
```

File: Dockerfile.chefdk (hunk 4 above)
Comment (olbrich) [reply Dockerfile.chefdk:14]: I think we can remove this one.

File: Dockerfile.cvp [hunk 8]
```diff
@@ -27,9 +27,12 @@ RUN apt-get update \
   curl \
   git \
   jq \
+  libcurl4-openssl-dev `# Needed to pull in libcurl4 for curl installation and enable to openssl TLS backend` \
+  libpq-dev `# Needed to provide libpq-fe.h header to build 'pg' gem native extensions` \
   software-properties-common \
   unzip \
   zip \
+  zlib1g-dev `# Fixes 'zlib is missing; necessary for building libxml2'` \
```
Comment (olbrich) [inline Dockerfile.cvp:35]:
```suggestion
  zlib1g-dev `# needed for 'nokogiri' gem` \
```

File: Dockerfile.quill [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+RUN apt-get update \
+  && apt-get install -y \
+  apt-transport-https \
+  build-essential \
+  ca-certificates \
+  curl \
+  git \
+  jq \
+  libcurl4-openssl-dev `# Needed to pull in libcurl4 for curl installation and enable to openssl TLS backend` \
+  libpq-dev `# Needed to provide libpq-fe.h header to build 'pg' gem native extensions` \
```
Comment (olbrich) [inline Dockerfile.quill:18]:
```suggestion
  libpq-dev `# Needed for 'pg' gem` \
```

File: Dockerfile.quill [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+  software-properties-common \
+  unzip \
+  zip \
+  zlib1g-dev `# Fixes 'zlib is missing; necessary for building libxml2'` \
```
Comment (olbrich) [inline Dockerfile.quill:22]:
```suggestion
  zlib1g-dev `# Needed for 'nokogiri' gem` \
```

**Review by olbrich — CHANGES_REQUESTED** [review]

**Review by Josephrfarmer — APPROVED** [review]
Approved based on Kevin's changes being addressed. 👍🏼

**Review by hedwards98 — APPROVED** [review]

**Review by olbrich — APPROVED** [review]

**Pull request merged**